v2/googlecloud-to-googlecloud/docs/BigtableChangeStreamsToBigQuery/README.md

```diff
@@ -0,0 +1,296 @@
+| timestamp | TIMESTAMP/INT | N | CBT cell’s timestamp. Type is determined by _writeNumericTimestamps_ pipeline option                                                                  |
+| timestamp_from | TIMESTAMP/INT | N | Time range start (inclusive) for a DeleteFromColumn mutation. Type is determined by _writeNumericTimestamps_ pipeline option                          |
+| timestamp_to | TIMESTAMP/INT | N | Time range end (exclusive) for a DeleteFromColumn mutation. Type is determined by _writeNumericTimestamps_ pipeline option                            |                             
+| value | STRING/BYTES | N | Bigtable cell value. Not specified for delete operations                                                                                              |
+| big_query_commit_timestamp* | TIMESTAMP | N | BQ auto-generated field describing the time when the record was written to BigQuery                                                                   |
+* Populating these columns can be skipped using pipeline options configuration. 
+
+## Getting started
+
+### Requirements
+* Java 11
+* Maven
+* Cloud Bigtable Instance exists
+* Cloud Bigtable table exists
+* Cloud Bigtable metadata Instance exists
+* Cloud Bigtable Change Streams feature is enabled for the Cloud Bigtable table
+* BigQuery dataset exists
+
+### Building Template
+This is a Flex Template meaning that the pipeline code will be containerized and the container will be
+run on Dataflow.
+
+##### Building Container Image
+
+* Set environment variables that will be used in the build process.
+* Run commands below from the repository root (DataflowTemplates).
+
+```sh
+export PROJECT=project
+export IMAGE_NAME=googlecloud-to-googlecloud
+export BUCKET_NAME=gs://bucket
+export TARGET_GCR_IMAGE=gcr.io/${PROJECT}/images/${IMAGE_NAME}
+export BASE_CONTAINER_IMAGE=gcr.io/dataflow-templates-base/java11-template-launcher-base
+export BASE_CONTAINER_IMAGE_VERSION=latest
+export TEMPLATE_MODULE=v2/googlecloud-to-googlecloud
+export COMMAND_MODULE=bigtable-changestreams-to-bigquery
+export APP_ROOT=/template/${COMMAND_MODULE}
+export COMMAND_SPEC=${APP_ROOT}/resources/${COMMAND_MODULE}-command-spec.json
+export TEMPLATE_IMAGE_SPEC=${BUCKET_NAME}/images/${COMMAND_MODULE}-image-spec.json
+```
+* Build and push image to Google Container Repository from the v2 directory
+
+```sh
+mvn clean package -Dimage=${TARGET_GCR_IMAGE} \
+                  -Dbase-container-image=${BASE_CONTAINER_IMAGE} \
+                  -Dbase-container-image.version=${BASE_CONTAINER_IMAGE_VERSION} \
+                  -Dapp-root=${APP_ROOT} \
+                  -Dcommand-spec=${COMMAND_SPEC} \
+                  -am -pl ${TEMPLATE_MODULE}
+```
+
+#### Creating Image Spec
+
+Create file in Cloud Storage with path to container image in Google Container Repository.
+
```

Comment:
We are not going to need the image spec anymore when using the plugin. It is hard for users to digest all the steps.

We can generate a set of instructions using `mvn package -DskipTests -PtemplatesSpec`, so in this section we could maybe point users to that auto generated doc (it's fine to have a more template-focused like this, though).

Reply:
Thanks for pointing this out. This is really helpful to reduce duplication and automate documentation generation. I reduced this doc to how to generate the docs and what the output schema looks like